src/main/java/bio/overture/songsearch/model/AnalysisFileFilter.java

```diff
@@ -0,0 +1,38 @@
+package bio.overture.songsearch.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import java.util.List;
+import lombok.Data;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AnalysisFileFilter {
+  private List<String> analysisTools;
+  private String fileType;
+  private String fileAccess;
+  private String dataType;
+
+  public boolean test(AnalysisFile file) {
+    if (fileAccess != null
+        && file.getFileAccess() != null
+        && !file.getFileAccess().equalsIgnoreCase(fileAccess)) {
+      return false;
+    }
+    if (fileType != null
+        && file.getFileType() != null
+        && !file.getFileType().equalsIgnoreCase(fileType)) {
+      return false;
+    }
+    if (dataType != null
+        && file.getDataType() != null
+        && !file.getDataType().equalsIgnoreCase(dataType)) {
+      return false;
+    }
+    if (analysisTools != null
+        && file.getAnalysisTools() != null
+        && !file.getAnalysisTools().containsAll(analysisTools)) {
+      return false;
+    }
```

Comment:
Am I reading correctly that if a file has `null` for the property being tested, the test will return `true`?

Reply:
hmm, I'm testing if the values in the filter match the values in the file. If any of them don't then its false, otherwise its true. But I see what your saying the check should be values don't match which currently won't include if the file values are null :thinking:.

Reply:
except i think your "containsAll" call is important, equals isn't perfect for the array value

Reply:
same with "equalsIgnoreCase" on the strings. I don't think there is a perfectly clean way to do this.

for the string matching its easier, just swap the order of variables in the equalsIgnoreCase condition:
```
(fileType != null
        && !fileType.equalsIgnoreCase(file.getFileType()))
```

but not easy for the array.

Reply:
This is the internet equivalent of talking over someone, my bad.

Reply:
Alright I've updated this if ladder. It should be correct now and a bit more cleaner.